internal/services/account_tokens.go

```diff
@@ -0,0 +1,515 @@
+		}
+	}()
+
+	latestCheckpointLedger, err := getLatestCheckpointLedger(s.archive)
+	if err != nil {
+		return err
+	}
+
+	log.Ctx(ctx).Infof("Populating account token cache from checkpoint ledger = %d", latestCheckpointLedger)
+	s.checkpointLedger = latestCheckpointLedger
+
+	// Create checkpoint change reader
+	reader, err := ingest.NewCheckpointChangeReader(ctx, s.archive, latestCheckpointLedger)
+	if err != nil {
+		return fmt.Errorf("creating checkpoint change reader: %w", err)
+	}
+	defer func() {
+		if closeErr := reader.Close(); closeErr != nil {
+			log.Ctx(ctx).Errorf("error closing checkpoint reader: %v", closeErr)
+		}
+	}()
+
+	// Collect account tokens from checkpoint
+	trustlines, contracts, contractTypesByContractID, contractIDsByWasmHash, err := s.collectAccountTokensFromCheckpoint(ctx, reader)
+	if err != nil {
+		return err
+	}
+
+	if err := s.enrichContractTypes(ctx, contractTypesByContractID, contractIDsByWasmHash); err != nil {
+		return err
+	}
+
+	return s.storeAccountTokensInRedis(ctx, trustlines, contracts, contractTypesByContractID)
+}
+
+// buildTrustlineKey constructs the Redis key for an account's trustlines set.
+func (s *accountTokenService) buildTrustlineKey(accountAddress string) string {
+	return s.trustlinesPrefix + accountAddress
+}
+
+// buildContractKey constructs the Redis key for an account's contracts set.
+func (s *accountTokenService) buildContractKey(accountAddress string) string {
+	return s.contractsPrefix + accountAddress
+}
+
+// buildContractTypeKey constructs the Redis key for a contract's type.
+func (s *accountTokenService) buildContractTypeKey(contractID string) string {
+	return s.contractTypePrefix + contractID
+}
+
+// Add adds token identifiers to an account's Redis set.
+// For trustlines, assets are formatted as "CODE:ISSUER".
+// For SAC balances, assets are contract addresses (C...).
+// Returns nil if assets is empty (no-op).
+func (s *accountTokenService) Add(ctx context.Context, redisKey string, assets []string) error {
+	if len(assets) == 0 {
+		return nil
+	}
+	if err := s.redisStore.SAdd(ctx, redisKey, assets...); err != nil {
+		return fmt.Errorf("adding members to set %s: %w", redisKey, err)
+	}
+	return nil
+}
+
+// GetAccountTrustlines retrieves all trustlines for an account from Redis.
+func (s *accountTokenService) GetAccountTrustlines(ctx context.Context, accountAddress string) ([]string, error) {
+	if accountAddress == "" {
+		return nil, fmt.Errorf("account address cannot be empty")
+	}
+	key := s.buildTrustlineKey(accountAddress)
+	trustlines, err := s.redisStore.SMembers(ctx, key)
+	if err != nil {
+		return nil, fmt.Errorf("getting trustlines for account %s: %w", accountAddress, err)
+	}
+	return trustlines, nil
+}
+
+// GetAccountContracts retrieves all Stellar Asset Contract (SAC) balance contract IDs for an account from Redis.
+func (s *accountTokenService) GetAccountContracts(ctx context.Context, accountAddress string) ([]string, error) {
```

Comment:
Is the idea here that this will basically be the contract id's for the list of trustlines the account has? Trying to think of the use-case where a wallet would prefer this over the trustlines. If I were to show my classic/sac balances, I would probably just want to use the account trustlines

Reply:
@piyalbasu My bad, I should have updated the func docstring with latest (Updated in https://github.com/stellar/wallet-backend/pull/369/commits/79ce45245aaea17ed6f719b1fa4bafd99a63de8a) . So `GetAccountContracts` will return the following:
1. G-address: all custom token IDs (non-SAC)
2. C-address: all custom and SAC token IDs

We will still call the `GetAccountTrustlines` for a G-address to fetch the classic assets